scripts/destroy_stage.ts

```diff
@@ -0,0 +1,234 @@
+import AWS from 'aws-sdk'
+
+AWS.config.update({
+    region: 'us-east-1',
+})
+
+const stackPrefixes = [
+    'app-web',
+    'app-api',
+    'ui-auth',
+    'uploads',
+    'postgres',
+    'storybook',
+    'infra-api',
+    'ui',
+]
+
+const stage = process.argv[2]
+
+const cf = new AWS.CloudFormation()
+const s3 = new AWS.S3()
+
+async function main() {
+    stackPrefixes.map(async (prefix) => {
+        const sn = `${prefix}-${stage}`
+        const clearBucketOutput = await clearServerlessDeployBucket(sn)
+        if (clearBucketOutput instanceof Error) {
+            console.log(clearBucketOutput)
+        }
+        const deleteStackOutput = await deleteStack(sn)
+        if (deleteStackOutput instanceof Error) {
+            console.log(deleteStackOutput)
+        }
+    })
+}
+
+interface s3ObjectKey {
+    Key: string
+    VersionId?: string
+}
+
+async function clearServerlessDeployBucket(
+    stackName: string
+): Promise<void | Error> {
+    // find all the buckets in our stack
+    const buckets = await getBucketsInStack(stackName)
+    if (buckets instanceof Error) {
+        return new Error(
+            `Could not get buckets in stack ${stackName}: ${buckets}`
+        )
+    }
+
+    // clean out each of those buckets
+    buckets.map(async (bucket) => {
+        // turn off versioning on the bucket
+        const versionResponse = await turnOffVersioningOnBucket(bucket)
+        if (versionResponse instanceof Error) {
+            return versionResponse
```

Comment:
I think these errors inside of buckets.map are not getting propagated up, just stored in the unused result of .map(), we probably want to surface an error eventually, (though we probably still want to try and keep destroying buckets if just one fails, too)

Reply:
Oh yup, forgot I was in a map here when sketching it. Good catch, I'll fix this.